src/query/service/src/pipelines/processors/transforms/window/transform_window.rs

```diff
@@ -734,6 +746,33 @@ where T: Number + ResultTypeOfUnary
         }
     }
 
+    fn count_rows(&self, start: &RowPtr, end: &RowPtr) -> usize {
+        if start.block == end.block {
+            end.row - start.row
+        } else {
+            let mut count = self.block_at(start).num_rows() - start.row + end.row;
+            for block in (start.block + 1)..end.block {
+                count += self.blocks[block].block.num_rows();
+            }
+            count
+        }
+    }
+
+    fn count_rows_in_frame(&mut self) {
+        self.frame_size = self.count_rows(&self.frame_start, &self.frame_end)
+    }
+
+    fn compute_on_frame(&mut self) -> Result<()> {
+        match &self.func {
+            WindowFunctionImpl::Aggregate(agg) => self.apply_aggregate(agg),
+            WindowFunctionImpl::PercentRank => {
+                self.count_rows_in_frame();
```

Comment:
There will not be reached if partition is not ended. We can add a assertion here, like `debug_assert!(self.partition_eneded`.